testsuite/features/step_definitions/command_steps.rb

```diff
@@ -179,6 +179,18 @@
   $command_output, _code = get_target('server').run(command)
 end
 
+When(/^I use spacewalk-common-channel to add all "([^"]*)" channels$/) do |os_product_version|
+  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, os_product_version)
+  raise ScriptError, "Synchronization error, version type #{os_product_version} in #{product} product not found" unless channels_to_synchronize
+
+  channels_to_synchronize.each do |os_product_version_channel|
+    log "Adding channel: #{os_product_version_channel}"
+    architecture = 'x86_64' # Only x86_64 is supported due to https://github.com/SUSE/spacewalk/issues/23663
```

Comment:
Ouch. I missed you need to pass them separated to the command. My bad.

I have been influenced by the SCC part where you pass them together, as a whole. Please accept my apologies.

However this is not true that you always is x86_64 . For example, as you noticed, it's `amd64-deb` for Debian or somthing like that.

I'm not sure about the solution. You could revert to your previous solution where you concatenate. Or you could introduce new arrays.

First solution is not valid for all "product names", like "fake", but it does not happen in your use case.

Second solution is overengineered.

I guess you initial solution was better, and you need to concatenate again. Sorry.

Reply:
No worries, I reverted this fragments to have the variable values explicitly defined by the user. I also created https://bugzilla.suse.com/show_bug.cgi?id=1219776 with regard to the discussion.